apps/src/blockly/addons/svgFrame.js

```diff
@@ -9,18 +9,22 @@ export default class SvgFrame {
   /**
    * Constructs an SvgFrame instance.
    * @param {Element} element - The block or workspace associated with the frame.
-   * @param {string} text - The text to display in the frame.
-   * @param {string} className - The CSS class name for styling.
-   * @param {string} textColor - The color for the frame's text.
-   * @param {string} headerColor - The color for the frame's header.
+   * @param {string} [text] - The text to display in the frame.
+   * @param {string} [className] - The CSS class name for styling.
+   * @param {Function} [getColor] - Get the color for the frame's header. This function should return a color value.
+   * @param {number} [headerHeight] - An optional override for the header size, used for workspace frames.
+   * @param {number} [fontSize] - An optional override for the size of the header text.
    */
-  constructor(element, text, className, textColor, headerColor) {
+  constructor(element, text, className, getColor, headerHeight, fontSize) {
     this.element_ = element;
     this.text = text || msg.block();
     this.className = className || 'svgFrame';
-    this.textColor = textColor || color.white;
-    this.headerColor = headerColor || color.light_gray;
-    this.baseColor = color.lightest_gray;
+    this.getColor = getColor;
+    this.headerColor = this.getColor() || color.light_gray;
+    this.baseColor = getBaseColor(this.headerColor);
+    this.textColor = color.white;
```

Comment:
are we guaranteed that our background will never be white, so white text will always work?

Reply:
Not technically. However, there are no white block styles because all blocks need to work with white text. The only way to create a frame with a white header would be by adding it in code, and I'd expect the engineer to see the problem with that right away.

Reply:
cool, this seems safe then!